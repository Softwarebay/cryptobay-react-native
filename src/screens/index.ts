export { HomeScreen } from './HomeScreen/HomeScreen';
export { WalthoughtScreen } from './WalthoughtScreen/WalthoughtScreen';
export { Screens } from './screens.enum';
