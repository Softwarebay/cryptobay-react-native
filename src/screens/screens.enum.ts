export enum Screens {
  HOME = 'Home',
  WALTHOUGHT = 'Walthought',
  SWAP = 'Swap',
  SETTINGS = 'Settings',
}
