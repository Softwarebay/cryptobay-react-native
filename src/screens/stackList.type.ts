export type StackList = {
  Home: undefined;
  Settings: undefined;
  Swap: undefined;
  WalletSetup: undefined;
  Walthought: undefined;
};
