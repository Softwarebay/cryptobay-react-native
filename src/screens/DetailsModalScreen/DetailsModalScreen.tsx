import { NativeStackScreenProps } from '@react-navigation/native-stack';
import React from 'react';

import { SwapDetails } from '../../components/organisms';
import { DetailsModalTemplate } from '../../components/templates';
import { tokensMock } from '../../mocks';
import { Screens } from '../../screens.enum';
import { StackList } from '../../types/stackList.type';
import { Token } from '../../types/token.type';

type DetailsModalScreenProps = NativeStackScreenProps<
  StackList,
  Screens.DETAILS
>;

export const DetailsModalScreen: React.FC<DetailsModalScreenProps> = ({
  route,
  navigation,
}: DetailsModalScreenProps) => {
  const { swapDetails } = route.params! as {
    swapDetails: SwapDetails;
  };

  const getTokenByShortName = () =>
    tokensMock.find(
      (foundToken: Token) => foundToken.shortName === swapDetails.shortName,
    ) || ({} as Token);

  const tokenShortName =
    getTokenByShortName() && getTokenByShortName().shortName;
  const tokenValue = getTokenByShortName() && getTokenByShortName().value;
  const tokenPrice = getTokenByShortName() && getTokenByShortName().price;

  const tokenValueInDolars = tokenValue * tokenPrice;
  const calculatedNetworkFee = (swapDetails.value * 0.09).toFixed(2);
  const totalAmount = `${tokenValue} ${tokenShortName}`;
  const totalAmountDolarValue = `$${tokenValueInDolars}`;
  const swapValue = `${swapDetails.value} ${tokenShortName}`;
  const networkFee = `${calculatedNetworkFee} ${tokenShortName}`;

  const onPressBack = () => navigation.goBack();

  return (
    <DetailsModalTemplate
      operation={swapDetails.operation}
      status={swapDetails.status}
      date={swapDetails.date}
      from={swapDetails.from}
      to={swapDetails.to}
      totalAmount={totalAmount}
      totalAmountDolarValue={totalAmountDolarValue}
      swapValue={swapValue}
      networkFee={networkFee}
      onPressBack={onPressBack}
    />
  );
};
