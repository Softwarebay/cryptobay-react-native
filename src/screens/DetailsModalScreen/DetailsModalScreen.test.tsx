import { fireEvent, render, screen } from '@testing-library/react-native';
import React from 'react';
import '@testing-library/jest-native';
import renderer from 'react-test-renderer';

import 'react-native';
import { Screens } from '../../screens.enum';

import { DetailsModalScreen } from './DetailsModalScreen';

const detailsModalScreenRoute = {
  key: '',
  name: Screens.DETAILS,
  params: {
    swapDetails: {
      shortName: 'BNB',
      operation: 'Received',
      status: 'Cancelled',
      date: 'Mar 3 at 10:04am',
      value: 0.04,
      from: '0x3Dc6...DfCE',
      to: '0x3Dc6...DfCE',
    },
  },
} as any;
const detailsModalScreenEmptyRoute = {
  key: '',
  name: Screens.DETAILS,
  params: {
    swapDetails: {},
  },
} as any;
const mockNavigation = {
  navigate: jest.fn(),
  goBack: jest.fn(),
} as any;

describe('DetailsModalScreen', () => {
  test('Component Snapshot', () => {
    const tree = renderer
      .create(
        <DetailsModalScreen
          navigation={mockNavigation}
          route={detailsModalScreenRoute}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('ensure finding token works correctly', () => {
    render(
      <DetailsModalScreen
        navigation={mockNavigation}
        route={detailsModalScreenRoute}
      />,
    );
    expect(
      screen.getByTestId('details-modal-template-total-amount'),
    ).toHaveTextContent('19.2371 BNB');
    expect(
      screen.getByTestId('details-modal-template-total-amount-dolars'),
    ).toHaveTextContent('$4360.858199');
    expect(
      screen.getByTestId('details-modal-template-swap-value'),
    ).toHaveTextContent('0.04 BNB');
  });

  test('ensure not found token works correctly', () => {
    render(
      <DetailsModalScreen
        navigation={mockNavigation}
        route={detailsModalScreenEmptyRoute}
      />,
    );
    expect(
      screen.getByTestId('details-modal-template-total-amount'),
    ).toHaveTextContent('');
    expect(
      screen.getByTestId('details-modal-template-total-amount-dolars'),
    ).toHaveTextContent('');
  });

  test('ensure onGoback works correctly', () => {
    render(
      <DetailsModalScreen
        navigation={mockNavigation}
        route={detailsModalScreenRoute}
      />,
    );
    fireEvent.press(screen.getByTestId('details-modal-template'));
    expect(mockNavigation.goBack).toHaveBeenCalled();
  });
});
