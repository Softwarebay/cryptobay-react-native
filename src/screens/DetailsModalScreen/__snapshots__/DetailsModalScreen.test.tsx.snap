// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsModalScreen Component Snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": "100%",
      "justifyContent": "flex-end",
      "width": "100%",
    }
  }
  testID="details-modal-template"
>
  <BlurView
    blurAmount={10}
    blurType="dark"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "alignItems": "center",
          "height": "40%",
          "justifyContent": "flex-end",
          "padding": 8,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#ABAFC4",
          "borderRadius": 4,
          "height": 4,
          "width": 40,
        }
      }
    />
  </BlurView>
  <View
    style={
      Object {
        "backgroundColor": "#17171A",
        "height": "60%",
        "padding": 24,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Poppins",
          },
          Object {
            "fontSize": 16,
            "fontWeight": "600",
            "lineHeight": 24,
          },
          Object {
            "alignSelf": "center",
            "color": "#FFFFFF",
            "marginBottom": 20,
          },
        ]
      }
    >
      Received
    </Text>
    <View
      style={
        Object {
          "marginBottom": 32,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Poppins",
                "fontSize": 12,
                "fontWeight": "400",
                "lineHeight": 18,
              },
              Object {
                "color": "#ABAFC4",
              },
              undefined,
            ]
          }
        >
          Status
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Poppins",
                "fontSize": 12,
                "fontWeight": "400",
                "lineHeight": 18,
              },
              Object {
                "color": "#ABAFC4",
              },
              undefined,
            ]
          }
        >
          Date
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Poppins",
                "fontSize": 14,
                "fontWeight": "600",
                "lineHeight": 24,
              },
              Object {
                "color": "#EA3943",
              },
              undefined,
            ]
          }
          testID="details-modal-status-badge"
        >
          Cancelled
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Poppins",
                "fontSize": 12,
                "fontWeight": "400",
                "lineHeight": 18,
              },
              Object {
                "color": "#ABAFC4",
              },
              undefined,
            ]
          }
        >
          Mar 3 at 10:04am
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "marginBottom": 32,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Poppins",
                "fontSize": 12,
                "fontWeight": "400",
                "lineHeight": 18,
              },
              Object {
                "color": "#ABAFC4",
              },
              undefined,
            ]
          }
        >
          From
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Poppins",
                "fontSize": 12,
                "fontWeight": "400",
                "lineHeight": 18,
              },
              Object {
                "color": "#ABAFC4",
              },
              undefined,
            ]
          }
        >
          To
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Poppins",
              },
              Object {
                "fontSize": 14,
                "fontWeight": "400",
                "lineHeight": 24,
              },
              Object {
                "color": "#FFFFFF",
              },
            ]
          }
        >
          0x3Dc6...DfCE
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Poppins",
              },
              Object {
                "fontSize": 14,
                "fontWeight": "400",
                "lineHeight": 24,
              },
              Object {
                "color": "#FFFFFF",
              },
            ]
          }
        >
          0x3Dc6...DfCE
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "borderColor": "#2F2F2F",
          "borderRadius": 10,
          "borderWidth": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "padding": 16,
            },
            Object {
              "borderBottomColor": "#2F2F2F",
              "borderBottomWidth": 1,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Poppins",
                },
                Object {
                  "fontSize": 14,
                  "fontWeight": "400",
                  "lineHeight": 24,
                },
                Object {
                  "color": "#FFFFFF",
                },
              ]
            }
          >
            Amount
          </Text>
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Poppins",
                },
                Object {
                  "fontSize": 14,
                  "fontWeight": "400",
                  "lineHeight": 24,
                },
                Object {
                  "color": "#FFFFFF",
                },
              ]
            }
            testID="details-modal-template-swap-value"
          >
            0.04 BNB
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Poppins",
                },
                Object {
                  "fontSize": 14,
                  "fontWeight": "400",
                  "lineHeight": 24,
                },
                Object {
                  "color": "#FFFFFF",
                },
              ]
            }
          >
            Network fee
          </Text>
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Poppins",
                },
                Object {
                  "fontSize": 14,
                  "fontWeight": "400",
                  "lineHeight": 24,
                },
                Object {
                  "color": "#FFFFFF",
                },
              ]
            }
          >
            0.00 BNB
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "padding": 16,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Poppins",
                },
                Object {
                  "fontSize": 16,
                  "fontWeight": "600",
                  "lineHeight": 24,
                },
                Object {
                  "color": "#FFFFFF",
                },
              ]
            }
          >
            Total Amount
          </Text>
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Poppins",
                },
                Object {
                  "fontSize": 16,
                  "fontWeight": "600",
                  "lineHeight": 24,
                },
                Object {
                  "color": "#FFFFFF",
                },
              ]
            }
            testID="details-modal-template-total-amount"
          >
            19.2371 BNB
          </Text>
        </View>
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Poppins",
                "fontSize": 12,
                "fontWeight": "400",
                "lineHeight": 18,
              },
              Object {
                "color": "#ABAFC4",
              },
              Object {
                "alignSelf": "flex-end",
                "flexDirection": "row",
              },
            ]
          }
          testID="details-modal-template-total-amount-dolars"
        >
          $4360.858199
        </Text>
      </View>
    </View>
  </View>
</View>
`;
